File: 1392-find-the-difference-of-two-arrays/1392-find-the-difference-of-two-arrays.py
class Solution:
    def findDifference(self, nums1: List[int], nums2: List[int]) -> List[List[int]]:
        set1 = set(nums1)
        set2 = set(nums2)
        common = set1 & set2
        return [list(set1 - common), list(set2 - common)]

File: 1392-find-the-difference-of-two-arrays/1392-find-the-difference-of-two-arrays.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[][]}
 */
var findDifference = function(nums1, nums2) {
    const set1 = new Set(nums1);
    const set2 = new Set(nums2);

    for (const n of set2) {
        if (set1.has(n)) {
            set1.delete(n);
            set2.delete(n);
        }
    }

    return [Array.from(set1), Array.from(set2)];
};